src/components/Tool/GoogleImport/GoogleImport.test.tsx

```diff
@@ -0,0 +1,368 @@
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { ApolloErgonoMockMap } from 'graphql-ergonomock';
+import { MockLinkCallHandler } from 'graphql-ergonomock/dist/apollo/MockLink';
+import { getSession } from 'next-auth/react';
+import { SnackbarProvider } from 'notistack';
+import TestRouter from '__tests__/util/TestRouter';
+import { GqlMockedProvider } from '__tests__/util/graphqlMocking';
+import { fetchAllData } from 'src/lib/deserializeJsonApi';
+import theme from 'src/theme';
+import GoogleImport from './GoogleImport';
+import { mockGoogleContactGroupsResponse } from './GoogleImportMocks';
+import { GoogleContactGroupsQuery } from './googleContactGroups.generated';
+
+const mockEnqueue = jest.fn();
+jest.mock('notistack', () => ({
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  ...jest.requireActual('notistack'),
+  useSnackbar: () => {
+    return {
+      enqueueSnackbar: mockEnqueue,
+    };
+  },
+}));
+
+jest.mock('src/lib/deserializeJsonApi');
+jest.mock('next-auth/react');
+
+const accountListId = 'account-id';
+const router = {
+  isReady: true,
+};
+
+const TestComponent = ({
+  mocks,
+  cache,
+  onCall,
+}: {
+  mocks: ApolloErgonoMockMap;
+  cache?: ApolloCache<object>;
+  onCall?: MockLinkCallHandler;
+}) => (
+  <SnackbarProvider>
+    <TestRouter router={router}>
+      <ThemeProvider theme={theme}>
+        <GqlMockedProvider<{
+          GoogleContactGroups: GoogleContactGroupsQuery;
+        }>
+          mocks={mocks}
+          cache={cache}
+          onCall={onCall}
+        >
+          <GoogleImport accountListId={accountListId} />
+        </GqlMockedProvider>
+      </ThemeProvider>
+    </TestRouter>
+  </SnackbarProvider>
+);
+
+describe('Google Import', () => {
+  const googleAccounts =
+    mockGoogleContactGroupsResponse.GoogleContactGroups.googleAccounts;
+  const account1Email = googleAccounts[0].email;
+  const account2Email = googleAccounts[1].email;
+  const account3Email = googleAccounts[2].email;
+  const account1GroupName = googleAccounts[0].contactGroups[0].title;
+  const account2GroupName = googleAccounts[1].contactGroups[0].title;
+  const account2GroupTag = googleAccounts[1].contactGroups[0].tag;
+
+  describe('render', () => {
+    it('multiple Google accounts', async () => {
+      const {
+        getByText,
+        findByText,
+        getAllByText,
+        getAllByRole,
+        queryByText,
+        getByRole,
+      } = render(
+        <TestComponent
+          mocks={{
+            GoogleContactGroups: {
+              ...mockGoogleContactGroupsResponse.GoogleContactGroups,
+            },
+          }}
+        />,
+      );
+      await waitFor(() => {
+        expect(getByText('Account to Import From')).toBeVisible();
+        expect(getAllByText(account1Email)[0]).toBeVisible();
+        expect(getByText(account1GroupName)).toBeVisible();
```

Comment:
I always prefer `await findBy...` instead of `await waitFor` + `getBy...` because it's simpler and has less noise. You should only need `waitFor` to wait for a function to be called or wait for an element to be removed from the DOM.